chainercv/extensions/detection/detection_vis_report.py

```diff
@@ -210,62 +78,30 @@ def __call__(self, trainer):
         if not _available:
             return
 
-        for idx in self.indices:
-            formated_filename_base = osp.join(trainer.out, self.filename_base)
-            out_file = (formated_filename_base +
-                        '_idx={}'.format(idx) +
-                        '_iter={}'.format(trainer.updater.iteration) + '.jpg')
-
-            inputs = self.dataset[idx]
-            self._check_type_dataset(inputs)
+        label_names = getattr(self.dataset, 'labels', None)
 
-            pred_bbox, pred_label = forward(self.target, inputs[0],
-                                            forward_func=self.predict_func,
-                                            expand_dim=True)
-            if pred_bbox.ndim == 2:
-                # force output to have batch axis
-                pred_bbox = pred_bbox[None]
-                pred_label = pred_label[None]
-            self._check_type_model((pred_bbox, pred_label))
-            pred_bbox = pred_bbox[0]  # (B, R, 5) -> (R, 5)
+        for idx in six.moves.range(len(self.dataset)):
+            out_file = self.filename.format(
+                index=idx, iteration=trainer.updater.iteration)
+            out_file = osp.join(trainer.out, out_file)
 
-            vis_transformed = self.vis_transform(inputs)
-            self._check_type_vis_transformed(vis_transformed)
-            vis_img = vis_transformed[0]
-            raw_bbox = vis_transformed[1]
+            img, gt_bbox, gt_label = self.dataset[idx]
```

Comment:
In my design, `predict` takes images in any sizes and returned bounding boxes are scaled to fit the original sizes.